File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
sum = 0
answer = 0

def solution(numbers, target):
    
    def dfs(idx):
        global sum, answer
        if idx == len(numbers):
            if sum == target:
                answer += 1
            return
       
        sum += numbers[idx]
        dfs(idx+1)
        sum -= numbers[idx]
    
        sum -= numbers[idx]
        dfs(idx+1)
        sum += numbers[idx]
    
    dfs(0)
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int cnt;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(numbers, target, 0, 0);
        return cnt;
    }
    public void dfs(int[] numbers, int target, int depth, int num){
        if(depth == numbers.length){
            if(num == target) cnt++;
            return;
        }
        dfs(numbers, target, depth+1, num + numbers[depth]);
        dfs(numbers, target, depth+1, num - numbers[depth]);   
    }
}